### PR TITLE
Exe icon implemented

### PR DESCRIPTION
**Pull request by SSPkrolik**
+ Added possibility to compile Windows resources into app executable.
+ Added appIconName setting to builder in order to compile app icon inside exe file.
+ Added Font directory deduction on Windows

### CODE REVIEW

File: nimx/naketools.nim [hunk 1]
```diff
@@ -177,6 +180,29 @@ proc makeMacOsBundle(b: Builder) =
     plist["NSHighResolutionCapable"] = %true
     plist.writePlist(bundlePath / "Contents" / "Info.plist")
 
+proc makeWindowsResource(b: Builder) =
+    let
+        rcPath = b.buildRoot / "res" / (b.appName & ".rc")
```
Comment (yglukhov) [inline nimx/naketools.nim]: please use origrespath.

File: nimx/naketools.nim [hunk 2]
```diff
@@ -177,6 +180,29 @@ proc makeMacOsBundle(b: Builder) =
...
+        rcO = b.nimcachePath / (b.appName & "_res.o")
+    var createResource: bool = false
+
+    shell "type", "nul", ">", rcPath
```
Comment (yglukhov) [inline nimx/naketools.nim]: can we not create a file in origrespath?

File: nimx/naketools.nim [hunk 3]
```diff
@@ -177,6 +180,29 @@ proc makeMacOsBundle(b: Builder) =
...
+
+    if not isNil(b.appIconName):
+        let appIconPath = b.resourcePath / (b.appIconName)
+
+        if fileExists(absPath(appIconPath)):
+            shell "echo", "AppIcon ICON \"$#\"" % [b.appIconName], ">>", rcPath
+            shell "windres", "-i", rcPath, "-o", rcO
+            createResource = true
+        else:
+            echo "Warning: icon was not found: $#" % [appIconPath]
```
Comment (yglukhov) [inline nimx/naketools.nim]:
```nim
echo "Warning: icon was not found: ", appIconPath
```
%)

**Comment by SSPkrolik** [comment]
Tested. Working on Windows.

**Pull request closed without merging**